flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecMultiJoin.java

```diff
@@ -0,0 +1,444 @@
+import org.apache.flink.table.planner.plan.nodes.exec.SingleTransformationTranslator;
+import org.apache.flink.table.planner.plan.nodes.exec.StateMetadata;
+import org.apache.flink.table.planner.plan.nodes.exec.utils.ExecNodeUtil;
+import org.apache.flink.table.planner.plan.nodes.exec.utils.TransformationMetadata;
+import org.apache.flink.table.planner.plan.utils.JoinUtil;
+import org.apache.flink.table.runtime.generated.GeneratedJoinCondition;
+import org.apache.flink.table.runtime.generated.JoinCondition;
+import org.apache.flink.table.runtime.generated.MultiJoinCondition;
+import org.apache.flink.table.runtime.operators.join.stream.StreamingMultiJoinOperator;
+import org.apache.flink.table.runtime.operators.join.stream.StreamingMultiJoinOperator.JoinType;
+import org.apache.flink.table.runtime.operators.join.stream.keyselector.AttributeBasedJoinKeyExtractor;
+import org.apache.flink.table.runtime.operators.join.stream.keyselector.JoinKeyExtractor;
+import org.apache.flink.table.runtime.operators.join.stream.utils.JoinInputSideSpec;
+import org.apache.flink.table.runtime.typeutils.InternalTypeInfo;
+import org.apache.flink.table.types.logical.RowType;
+
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonCreator;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonInclude;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonProperty;
+
+import org.apache.calcite.rex.RexNode;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.apache.flink.util.Preconditions.checkArgument;
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * Stream {@link StreamExecNode} for N-way Joins. This node handles multi-way joins in streaming
+ * mode, supporting different join types and conditions for each input.
+ */
+@ExecNodeMetadata(
+        name = "stream-exec-multi-join",
+        version = 1,
+        producedTransformations = StreamExecMultiJoin.MULTI_JOIN_TRANSFORMATION,
+        minPlanVersion = FlinkVersion.v2_0,
+        minStateVersion = FlinkVersion.v2_0)
+public class StreamExecMultiJoin extends ExecNodeBase<RowData>
+        implements StreamExecNode<RowData>, SingleTransformationTranslator<RowData> {
+
+    public static final String MULTI_JOIN_TRANSFORMATION = "multi-join";
+
+    // Field names for JSON serialization
+    private static final String FIELD_NAME_ID = "id";
+    private static final String FIELD_NAME_TYPE = "type";
+    private static final String FIELD_NAME_CONFIGURATION = "configuration";
+    private static final String FIELD_NAME_INPUT_PROPERTIES = "inputProperties";
+    private static final String FIELD_NAME_OUTPUT_TYPE = "outputType";
+    private static final String FIELD_NAME_DESCRIPTION = "description";
```

Comment:
Those can be removed. They are given by the upper interface already.Those can be removed. They are given by the upper interface already.

Reply:
done